In the United States, the cost of university is extremely high. A large part of the money universities collect every year is spent on salaries.

## But who get the most money?

Let's take a look at the data from Pennsylvanian Universities

In [5]:
import pandas as pa     #linear algebra
import numpy as nu      #data processing
import os

In [6]:
pwd = os.getcwd()

In [7]:
data_df = pa.read_csv( pwd + '//2013_Salaries__Pennsylvania_State_System_of_Higher_Education.csv')       #reading the file
data_df.head(6)     #print the first (6)

index                     Name Base Pay University or Office  \
0      0  Driscoll, Michael Allan  $275000              Indiana   
1      1      Weisenstein, Greg R  $241935         West Chester   
2      2          Welsh, Marcia L  $230000     East Stroudsburg   
3      3        Norton, Cheryl J.  $225000        Slippery Rock   
4      4        Wollman, Julie E.  $225000             Edinboro   
5      5      Mcnairy, Francine G  $222100         Millersville   

               Position  
0  University President  
1  University President  
2  University President  
3  University President  
4  University President  
5  University President

In [8]:
data_df.shape       #The shape of the data

(12632, 5)

In [9]:
data_df.describe()      #describe the data

index
count  12632.000000
mean    6315.500000
std     3646.688635
min        0.000000
25%     3157.750000
50%     6315.500000
75%     9473.250000
max    12631.000000

The describe not corrct becose the data is objict , so we will change it :

In [10]:
data_df.nunique()       #unique value of columns 

index                   12632
Name                    12583
Base Pay                 2126
University or Office       15
Position                 3315
dtype: int64

so here we can found The unique value of columns 
There 12632 index
## 12583       Name     
## 2126        Base Pay
## 15          University or Office
## 3315        Position                               

In [11]:
data_df.notnull().value_counts()        #is there a not.nan values in the data?

index  Name  Base Pay  University or Office  Position
True   True  True      True                  True        12632
dtype: int64

In [12]:
data_df.info()      #the data info

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12632 entries, 0 to 12631
Data columns (total 5 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   index                 12632 non-null  int64 
 1   Name                  12632 non-null  object
 2   Base Pay              12632 non-null  object
 3   University or Office  12632 non-null  object
 4   Position              12632 non-null  object
dtypes: int64(1), object(4)
memory usage: 493.6+ KB


In [13]:
data_df

index                     Name Base Pay      University or Office  \
0          0  Driscoll, Michael Allan  $275000                   Indiana   
1          1      Weisenstein, Greg R  $241935              West Chester   
2          2          Welsh, Marcia L  $230000          East Stroudsburg   
3          3        Norton, Cheryl J.  $225000             Slippery Rock   
4          4        Wollman, Julie E.  $225000                  Edinboro   
...      ...                      ...      ...                       ...   
12627  12627      Black, Jacqueline D    $1254                  Edinboro   
12628  12628             Webster, Kim    $1245                  Kutztown   
12629  12629       Hanelly, William T  $152950                Lock Haven   
12630  12630       Moran, James D Iii  $193125  Office of the Chancellor   
12631  12631        Cavanaugh, John C  $327500  Office of the Chancellor   

                                         Position  
0                            University President  
1                            University President  
2                            University President  
3                            University President  
4                            University President  
...                                           ...  
12627                         Instructor, PT Temp  
12628                                  Instructor  
12629                    Vice President/Fin.Admin  
12630  Vice Chancellor Academic & Student Affairs  
12631                                  Chancellor  

[12632 rows x 5 columns]

## So we chack the data and now lets clean it

1- we will delete the 'index' column cuz we dont need it

2- change the data type of 'Base Pay' to flout , and delete the symbol '$'

In [14]:
#Delete the index column
data_df.drop(columns=['index'] , inplace=True)

In [15]:
data_df['Base Pay'].value_counts()      #we need to delete $

$107870    764
$88745     458
$44795     365
$11199     341
$27012     234
          ... 
$71070       1
$71087       1
$71105       1
$71201       1
$327500      1
Name: Base Pay, Length: 2126, dtype: int64

In [16]:
data_df['Base Pay'] = data_df['Base Pay'].str.replace('$', '', regex=False)     #replaece the remove $ and set the '' 

In [17]:
data_df['Base Pay'].dtypes

dtype('O')

In [18]:
data_df['Base Pay'] = data_df['Base Pay'].astype('float')       #change the datatype to flout

In [19]:
data_df['Base Pay'].dtypes

dtype('float64')

In [20]:
data_df

Name  Base Pay      University or Office  \
0      Driscoll, Michael Allan  275000.0                   Indiana   
1          Weisenstein, Greg R  241935.0              West Chester   
2              Welsh, Marcia L  230000.0          East Stroudsburg   
3            Norton, Cheryl J.  225000.0             Slippery Rock   
4            Wollman, Julie E.  225000.0                  Edinboro   
...                        ...       ...                       ...   
12627      Black, Jacqueline D    1254.0                  Edinboro   
12628             Webster, Kim    1245.0                  Kutztown   
12629       Hanelly, William T  152950.0                Lock Haven   
12630       Moran, James D Iii  193125.0  Office of the Chancellor   
12631        Cavanaugh, John C  327500.0  Office of the Chancellor   

                                         Position  
0                            University President  
1                            University President  
2                            University President  
3                            University President  
4                            University President  
...                                           ...  
12627                         Instructor, PT Temp  
12628                                  Instructor  
12629                    Vice President/Fin.Admin  
12630  Vice Chancellor Academic & Student Affairs  
12631                                  Chancellor  

[12632 rows x 4 columns]

so lets check the describe()

In [21]:
data_df.describe()

Base Pay
count   12632.000000
mean    57575.800823
std     29768.238649
min      1245.000000
25%     34328.000000
50%     51856.000000
75%     76661.000000
max    327500.000000

## so, The main Q was , who gets the most money?

- The person who gets the most money

- The Position that's gets the most money

In [22]:
data_df[[ 'Name','Base Pay','Position']].sort_values(by=['Base Pay'] , ascending=False)[:30]      #sort the data by Base pay

Name  Base Pay  \
12631           Cavanaugh, John C  327500.0   
0         Driscoll, Michael Allan  275000.0   
1             Weisenstein, Greg R  241935.0   
2                 Welsh, Marcia L  230000.0   
3               Norton, Cheryl J.  225000.0   
4               Wollman, Julie E.  225000.0   
5             Mcnairy, Francine G  222100.0   
6          Fiorentino, Michael Jr  220000.0   
7               Whitney, Karen M.  220000.0   
8      Cevallos, Francisco Javier  218956.0   
9           Vargas-Aburto, Carlos  207568.0   
10                 Soltz, David L  207197.0   
11               Ruud, William N.  206196.0   
12               Nowaczyk, Ronald  205000.0   
13       Howard-Vital, Michelle R  202430.0   
14             Jones, Geraldine M  200820.0   
15                Prabhu, Vilas A  200366.0   
16               Garland, Peter H  199477.0   
17                  Golden, Allan  198956.0   
18            Williams, William F  195118.0   
12630          Moran, James D Iii  193125.0   
19             Intemann, Gerald W  192610.0   
21               Eckert, Gerald C  190668.0   
20            Bruszewski, Roger V  190668.0   
22                Dillon, James S  187204.0   
23               Dent, Gary Kever  183799.0   
24                 Tripp, Harry E  182900.0   
25                   Blake, Ira K  182000.0   
26              Wooten, Cornelius  182000.0   
27              Staneski, Richard  181847.0   

                                         Position  
12631                                  Chancellor  
0                            University President  
1                            University President  
2                            University President  
3                            University President  
4                            University President  
5                            University President  
6                            University President  
7                            University President  
8                            University President  
9                   Provost & VP Academic Affairs  
10                           University President  
11                           University President  
12              Provost & VP for Academic Affairs  
13                           University President  
14                    Acting University President  
15       Provost & Vice Pres for Academic Affairs  
16                      Executive Vice Chancellor  
17                   Interim University President  
18       Provost & Vice President Academic Affair  
12630  Vice Chancellor Academic & Student Affairs  
19                  Provost & VP Academic Affairs  
21                   Vice Pres for University Adv  
20                   Vice Pres for Finance & Admn  
22                  VC-Administration and Finance  
23                    Vice Chancellor for HR & LR  
24                   VP of Student & Univ Affairs  
25                    Provost/VP Academic Affairs  
26                             VP Admin & Finance  
27                    VP Finance & Administration

## So the person who get the most money is : Cavanaugh, John C ,AND his base Pay is : 327500.0 , AND he is a Chancellor

In [23]:
data_df.Position.value_counts().head(7)

Assistant Professor    1196
Associate Professor    1167
Professor              1077
Instructor              851
Custodial Worker 1      432
Clerk Typist 2          344
Clerk Typist 3          172
Name: Position, dtype: int64

In [24]:
data_Assistant = data_df[data_df.Position == 'Assistant Professor' ]
data_Assistant.sum()

Name                    Gualardo, Samuel JLeffert, John E IvLueken, Pa...
Base Pay                                                       76198199.0
University or Office    IndianaLock HavenSlippery RockLock HavenClario...
Position                Assistant ProfessorAssistant ProfessorAssistan...
dtype: object

In [25]:
data_Associate = data_df[data_df.Position == 'Associate Professor' ]
data_Associate.sum()

Name                    Holland, Carol LSmith, Anna MaeFreeman, Richar...
Base Pay                                                       95210182.0
University or Office    Slippery RockLock HavenLock HavenIndianaLock H...
Position                Associate ProfessorAssociate ProfessorAssociat...
dtype: object

In [26]:
data_Professor = data_df[data_df.Position == 'Professor' ]
data_Professor.value_counts()

Name                     Base Pay  University or Office  Position 
Abeysinghe, Wadduwage P  102672.0  Lock Haven            Professor    1
Perner, Darlene E        102672.0  Bloomsburg            Professor    1
Patrick, John            97783.0   California            Professor    1
Patrick, Kevin J         100228.0  Indiana               Professor    1
Patte, Michael M         88692.0   Bloomsburg            Professor    1
                                                                     ..
Hamilton, Sharon A.      107870.0  Edinboro              Professor    1
Hamrick, John M          107870.0  Kutztown              Professor    1
Hanauer, David Ian       95455.0   Indiana               Professor    1
Hanich, Laurie B         84469.0   Millersville          Professor    1
Zuraikat, Nashat M       107870.0  Indiana               Professor    1
Length: 1077, dtype: int64

In [27]:
data_Instructor = data_df[data_df.Position == 'Instructor' ]
data_Instructor.sum()

Name                    Fitting, Gary DavidPike, Lynn AnnRupert, Marth...
Base Pay                                                       25352734.0
University or Office    IndianaIndianaIndianaIndianaIndianaIndianaIndi...
Position                InstructorInstructorInstructorInstructorInstru...
dtype: object

In [28]:
data_Custodial = data_df[data_df.Position == 'Custodial Worker 1' ]
data_Custodial.sum()

Name                    Anderson, Walter CCooper, Barbara EDonahue, Ke...
Base Pay                                                       11723668.0
University or Office    CaliforniaWest ChesterIndianaEast StroudsburgI...
Position                Custodial Worker 1Custodial Worker 1Custodial ...
dtype: object

In [29]:
data_Clerk = data_df[data_df.Position == 'Clerk Typist 2' ]
data_Clerk.sum()

Name                    Bard, Jill RBowser, Joan KDubbs, Lisa MFranken...
Base Pay                                                       10431099.0
University or Office    ShippensburgIndianaShippensburgSlippery RockSh...
Position                Clerk Typist 2Clerk Typist 2Clerk Typist 2Cler...
dtype: object

## So the Position gets the most money is : Professors ,AND their base Pay is : 110,591,809.0 , AND they are : 1077 Professors.

_____________________________________________________________________________________________________

## our work is done here , but lets see how much can we get information from this data

In [30]:
data_df[0:50]

Name  Base Pay      University or Office  \
0      Driscoll, Michael Allan  275000.0                   Indiana   
1          Weisenstein, Greg R  241935.0              West Chester   
2              Welsh, Marcia L  230000.0          East Stroudsburg   
3            Norton, Cheryl J.  225000.0             Slippery Rock   
4            Wollman, Julie E.  225000.0                  Edinboro   
5          Mcnairy, Francine G  222100.0              Millersville   
6       Fiorentino, Michael Jr  220000.0                Lock Haven   
7            Whitney, Karen M.  220000.0                   Clarion   
8   Cevallos, Francisco Javier  218956.0                  Kutztown   
9        Vargas-Aburto, Carlos  207568.0                  Kutztown   
10              Soltz, David L  207197.0                Bloomsburg   
11            Ruud, William N.  206196.0              Shippensburg   
12            Nowaczyk, Ronald  205000.0                   Clarion   
13    Howard-Vital, Michelle R  202430.0                   Cheyney   
14          Jones, Geraldine M  200820.0                California   
15             Prabhu, Vilas A  200366.0              Millersville   
16            Garland, Peter H  199477.0  Office of the Chancellor   
17               Golden, Allan  198956.0                 Mansfield   
18         Williams, William F  195118.0             Slippery Rock   
19          Intemann, Gerald W  192610.0                   Indiana   
20         Bruszewski, Roger V  190668.0              Millersville   
21            Eckert, Gerald C  190668.0              Millersville   
22             Dillon, James S  187204.0  Office of the Chancellor   
23            Dent, Gary Kever  183799.0  Office of the Chancellor   
24              Tripp, Harry E  182900.0                   Clarion   
25                Blake, Ira K  182000.0                Bloomsburg   
26           Wooten, Cornelius  182000.0                   Indiana   
27           Staneski, Richard  181847.0          East Stroudsburg   
28   Fiorentino, Christopher M  180883.0              West Chester   
29               Ball, Karen S  180250.0  Office of the Chancellor   
30            Lyman, Barbara G  180030.0              Shippensburg   
31              Lamwers, Linda  178879.0              West Chester   
32         Silberman, Gerald L  178048.0                  Kutztown   
33            Reidhead, Van A.  177000.0          East Stroudsburg   
34            Curry, Charles T  176474.0             Slippery Rock   
35           Ebbs, George H Jr  176025.0                Bloomsburg   
36              Camp, Robert C  175004.0                   Indiana   
37             Wilson, Donna F  175000.0                Lock Haven   
38        Snavely, Deanne Lynn  172500.0                   Indiana   
39      Harpster, George F. Jr  172458.0              Shippensburg   
40                Mixner, Mark  171981.0              West Chester   
41              Ruth, Rickey E  171198.0              Shippensburg   
42            Herbst, Gordon J  168152.0                  Edinboro   
43         Rugen, Richard Hall  167197.0                Bloomsburg   
44                 Banks, Ivan  166860.0                   Cheyney   
45      Dempsey, William A Iii  166800.0                  Kutztown   
46             Pavlovich, Mark  165456.0              West Chester   
47        Bricketto, Matthew J  164158.0              West Chester   
48            Luckey, Rhonda H  163584.0                   Indiana   
49               Koch, Linda D  163129.0                Lock Haven   

                                    Position  
0                       University President  
1                       University President  
2                       University President  
3                       University President  
4                       University President  
5                       University President  
6                       University President  
7                       University President  
8                       Universi

- we can find the univarsty that get the most pepole
- we can find the univarsty that get the least pepole
- we can find the univarsty that payed the most 
- we can find the persone that payed the least
- we can find the position that payed the least
- top 10 names ber sallary
- dawn 10 names ber sallary
- top 10 University or Office ber sallary

In [31]:
data_df[['University or Office']].value_counts()[0:20]

University or Office    
West Chester                1628
Indiana                     1501
Bloomsburg                  1096
Millersville                1022
Kutztown                     990
Slippery Rock                909
Shippensburg                 889
California                   884
Edinboro                     806
Clarion                      740
East Stroudsburg             719
Lock Haven                   566
Mansfield                    439
Cheyney                      287
Office of the Chancellor     156
dtype: int64

## the University or Office that get the most pepole is : West Chester , and the get less is : Office of the Chancellor

In [32]:
data_df[data_df["University or Office"] == "West Chester"].sum()

Name                    Weisenstein, Greg RFiorentino, Christopher MLa...
Base Pay                                                       89830456.0
University or Office    West ChesterWest ChesterWest ChesterWest Chest...
Position                University PresidentDean-Bus & Pub AffairsProv...
dtype: object

In [33]:
data_df[data_df["University or Office"] == "Indiana"].sum() 

Name                    Driscoll, Michael AllanIntemann, Gerald WWoote...
Base Pay                                                       90714032.0
University or Office    IndianaIndianaIndianaIndianaIndianaIndianaIndi...
Position                University PresidentProvost & VP Academic Affa...
dtype: object

In [34]:
data_df[data_df["University or Office"] == "Bloomsburg"].sum() 

Name                    Soltz, David LBlake, Ira KEbbs, George H JrRug...
Base Pay                                                       61535705.0
University or Office    BloomsburgBloomsburgBloomsburgBloomsburgBlooms...
Position                University PresidentProvost/VP Academic Affair...
dtype: object

## So the university, that gets the most money is: Indiana / and they spend : 90,714,032.0 mil to the sallary

In [35]:
data_df[[ 'Name','Base Pay','Position']].sort_values(by=['Base Pay'] , ascending=False).tail(10)     #sort the data by Base pay

Name  Base Pay                   Position
12619     Williams, Todd D    2491.0                 Instructor
12621    Scandrett, Lucy C    2464.0                 Instructor
12620      Grice, Wendy A.    2464.0             Instructor MUS
12622       Ekberg, Dean W    2240.0  Temp. PT Instructor/Music
12623    Miller, Shelley M    2006.0    Cheerleading Co-Adviser
12624       Cordle, Adam P    1411.0            Instructor-Temp
12625  Staherski, Cheryl E    1373.0            Instructor-RPTF
12626   Shaginaw, Justin W    1344.0                 Instructor
12627  Black, Jacqueline D    1254.0        Instructor, PT Temp
12628         Webster, Kim    1245.0                 Instructor

## the persone that payed the least is : Webster, Kim ,and his payment is : 1245.0 , and the Position that get less base pay is : Instructor

In [36]:
data_df[[ 'Name','Base Pay','Position']].sort_values(by=['Base Pay'] , ascending=False)    #sort the data by Base pay

Name  Base Pay              Position
12631        Cavanaugh, John C  327500.0            Chancellor
0      Driscoll, Michael Allan  275000.0  University President
1          Weisenstein, Greg R  241935.0  University President
2              Welsh, Marcia L  230000.0  University President
3            Norton, Cheryl J.  225000.0  University President
...                        ...       ...                   ...
12624           Cordle, Adam P    1411.0       Instructor-Temp
12625      Staherski, Cheryl E    1373.0       Instructor-RPTF
12626       Shaginaw, Justin W    1344.0            Instructor
12627      Black, Jacqueline D    1254.0   Instructor, PT Temp
12628             Webster, Kim    1245.0            Instructor

[12632 rows x 3 columns]

## top 10 names ber sallary are :Cavanaugh, John C,     Driscoll, Michael Allan,     Weisenstein, Greg R,    Welsh, Marcia L,    Norton, Cheryl J.,      Wollman, Julie E.,         Mcnairy, Francine G,        Fiorentino, Michael Jr,      Whitney, Karen M.,      Cevallos, Francisco Javier . And the 90% of them are University President

In [37]:
data_df[[ 'Name','Base Pay','Position']].sort_values(by=['Base Pay'] , ascending=True)[:10]  #sort the data by Base pay

Name  Base Pay                   Position
12628              Webster, Kim    1245.0                 Instructor
12627       Black, Jacqueline D    1254.0        Instructor, PT Temp
12626        Shaginaw, Justin W    1344.0                 Instructor
12625       Staherski, Cheryl E    1373.0            Instructor-RPTF
12624            Cordle, Adam P    1411.0            Instructor-Temp
12623         Miller, Shelley M    2006.0    Cheerleading Co-Adviser
12622            Ekberg, Dean W    2240.0  Temp. PT Instructor/Music
12621         Scandrett, Lucy C    2464.0                 Instructor
12620           Grice, Wendy A.    2464.0             Instructor MUS
12618  Schottman, Margaret Love    2491.0        Instructor, PT Temp

In [38]:
data_df[['Base Pay', 'University or Office']].sort_values(by=['Base Pay'] , ascending=False).head(10)  #sort the data by Base pay

Base Pay      University or Office
12631  327500.0  Office of the Chancellor
0      275000.0                   Indiana
1      241935.0              West Chester
2      230000.0          East Stroudsburg
3      225000.0             Slippery Rock
4      225000.0                  Edinboro
5      222100.0              Millersville
6      220000.0                Lock Haven
7      220000.0                   Clarion
8      218956.0                  Kutztown

## top 10 University or Office ber sallary are : Office of the Chancellor, Indiana, West Chester, East Stroudsburg, Slippery Rock, Edinboro, Millersville, Lock Haven, Clarion, Kutztown